### DISCUSSION TITLE
bug :wired_table_rec/table_recover.py

### DISCUSSION THREAD

**Discussion by Kyrie666** [RapidAI/TableStructureRec#33] [Q&A] [answered]
def get_rows(polygons: np.array)  这个方法： if max(split_idxs) != len(minus_res): 出现ValueError: max() arg is an empty sequence，我看了一下写法，确实有异常，需要处理

**Comment by Kyrie666** [comment, marked as answer]
split_idxs = np.argwhere(abs(minus_res) > thresh).squeeze() 输出变量是一个数组，可能是空数组，但仍然是一个数组；
if split_idxs.ndim == 0: 这个条件永远不会为真，max(split_idxs) 就会报错。

**Reply by Joker1212** [reply]
能否留一下出现这个异常的图片呢？

**Reply by Kyrie666** [reply]
图片我找到了，但是目前我的开发环境与网络环境是隔离，我无法将图片上传给你。我可以描述一下：是pdf的目录页，但是底部多了一个框，里面是个签名。算法应该是在英文这个框，判断使用wired方案。

**Reply by Joker1212** [reply]
非常感谢指出这个问题，我修一下这里

**Reply by Joker1212** [reply]
但是这里为空，其实相当于完全失去了表格识别的功能，提取目录的话，建议直接进行ocr识别会更好，担心排版问题的话，可以根据ocr框的间距来处理文字间距

**Comment by Kyrie666** [comment]
OK，我帮你测试一下



---Original---
From: ***@***.***&gt;
Date: Fri, Sep 27, 2024 14:56 PM
To: ***@***.***&gt;;
Cc: "Kyrie ***@***.******@***.***&gt;;
Subject: Re: [RapidAI/TableStructureRec] bug :wired_table_rec/table_recover.py(Discussion #33)




 
能否留一下出现这个异常的图片呢？

**Comment by Kyrie666** [comment]
其实这里对我的功能不太影响，我的场景是处理pdf文档，里面含有大量的的表格！我没有单独对某一页进行针对性的处理！只是偶然遇到这个问题，交流一下，谢谢



---Original---
From: ***@***.***&gt;
Date: Fri, Sep 27, 2024 15:19 PM
To: ***@***.***&gt;;
Cc: "Kyrie ***@***.******@***.***&gt;;
Subject: Re: [RapidAI/TableStructureRec] bug :wired_table_rec/table_recover.py(Discussion #33)




 
但是这里为空，其实相当于完全失去了表格识别的功能，提取目录的话，建议直接进行ocr识别会更好，担心排版问题的话，可以根据ocr框的间距来处理文字间距
